jwala-common/src/main/java/com/cerner/jwala/common/rule/group/GroupNameRule.java

```diff
@@ -16,8 +17,40 @@ protected MessageResponseStatus getMessageResponseStatus() {
         return FaultType.INVALID_GROUP_NAME;
     }
 
+
+    protected MessageResponseStatus getMessageResponseGroupLengthStatus() { return FaultType.GROUP_NAME_TOO_LONG; }
+
     @Override
     protected String getMessage() {
         return "Invalid Group Name: \"" + name + "\"";
     }
+
+    protected String getMessageGroupLength(){ return "Group name is too long. Please enter a Group name lesser than 256 characters";}
+
+    @Override
+    public void validate() throws BadRequestException {
+        if (!isValid()) {
+            throw new BadRequestException(getMessageResponseStatus(),
+                    getMessage());
+        }
+        if(!isValidLength()){
+            throw new BadRequestException(getMessageResponseGroupLengthStatus(),getMessageGroupLength());
+        }
+    }
+
+
+    /**
+     * Checks for null
+     * Checks for an empty string +only spaces
+     * checks for a string with last character space
+     * checks for a string with first character space
+     * checks that the string only contains alphanumeric+period+dash+underscore+space
+     *
+     * @return boolean
+     */
+    @Override
+    public boolean isValid() {return name != null && !"".equals(name.trim()) && !name.matches(".*[\\s]$") && !name.matches("^[\\s].*") && name.matches("[A-Za-z0-9._\\s-]+");
```

Comment:
All the three rules (WebServer, Group and JVM have the same pattern. Can you push this to the ValidNameRule baseclass?